suite-native/state/src/store.ts

```diff
@@ -18,6 +18,7 @@ const middlewares: Middleware[] = [
     prepareBlockchainMiddleware(extraDependencies),
     prepareFiatRatesMiddleware(extraDependencies),
     logsMiddleware,
+    // Caution! the Discovery middleware must be before the Device middleware to work properly.
```

Comment:
why?

Reply:
I've moved part of the device-related authentication logic from `discoveryMiddleware` to `deviceMiddleware` where it belongs. Since the way the discovery reacts to actions is depending on it's evaluation order, the `discoveryMiddleware` has to be defined before the `deviceMiddleware`.

Reply:
That kind of sucks, what are other possible solutions to avoid these hacks or something?

Reply:
Ok, I was able to resolve it. The problem was that while passing action in `discoveryMiddleware` with the `next` function, it wasn't waiting for its result. I prepended the `next(action)` with `await` and now it works as expected independently on the order of the middlewares. Fixed in 41afd8c4f.

Reply:
I think it would still be nice to write a comment and explain exactly why there is async and which thunk is being awaited. This will prevent that someone will accidentally remove it like we did just now.

Reply:
So... I've done one more round of refactoring and the last solution is not order-dependent nor use async middleware.

I've encapsulated the createDiscovery and startDiscovery to a new thunk that handles this asynchronous dependency to simplify the middleware.

688afbad1